src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -1018,23 +1046,33 @@ _xccdf_policy_rule_evaluate(struct xccdf_policy * policy, const struct xccdf_rul
 	const char *message = NULL;
 	int report = 0;
 
-	/* If policy selects only one rule and the rule currently being
-	 * evaluated is not equal to the selected rule, do not evaluate it and
-	 * mark it as notselected. */
-	if (policy->rule != NULL) {
-		if (strcmp(policy->rule, rule_id) != 0) {
+	/* If the rule is requested to be skipped by the user using --skip-rule on
+	 * the command line we will skip the evaluation of this rule. */
+	if (oscap_htable_get(policy->skip_rules, rule_id) != NULL) {
+		return _xccdf_policy_report_rule_result(policy, result, rule, NULL, XCCDF_RESULT_NOT_SELECTED, NULL);
+	}
+
+	/* If user wants to evaluate only specific rules and the rule currently
+	 * being evaluated is not among these rules, do not evaluate it and mark it
+	 * as notselected. */
+	if (_user_specified_rule_mode(policy) > 0) {
```

Comment:
Or use `oscap_htable_itemcount(policy->rules) > 0` here and get rid of that function.